cpp/src/parquet/column_reader_benchmark.cc

```diff
@@ -0,0 +1,123 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "benchmark/benchmark.h"
+#include "parquet/column_page.h"
+#include "parquet/column_reader.h"
+#include "parquet/schema.h"
+#include "parquet/test_util.h"
+#include "parquet/types.h"
+
+namespace parquet {
+
+using benchmark::DoNotOptimize;
+using parquet::Repetition;
+using parquet::test::MakePages;
+using schema::NodePtr;
+
+namespace benchmark {
+
+// Benchmarks Skip and ReadBatch API for ColumnReader with the following
+// paramenters in order:
+// - def_level: set to 0 for REQUIRED, 1 for OPTIONAL/REPEATED.
+// - rep_level: set to 1 for REPEATED, 0 otherwise.
+// - is_skip: set to 0 for benchmarking ReadBatch and 1 for Skip.
+// - batch_size: sets how many values to read at each call.
+static void BM_Skip(::benchmark::State& state) {
+  internal::LevelInfo level_info;
+  level_info.def_level = state.range(0);
+  level_info.rep_level = state.range(1);
+  const int skip = state.range(2);
+  const int batch_size = state.range(3);
+
+  Repetition::type repetition = Repetition::REQUIRED;
+  if (level_info.def_level > 0) {
+    repetition = Repetition::OPTIONAL;
+  }
+  if (level_info.rep_level > 0) {
+    repetition = Repetition::REPEATED;
+  }
+  NodePtr type = schema::Int32("b", repetition);
+  const ColumnDescriptor descr(type, level_info.def_level, level_info.rep_level);
+
+  const int num_pages = 5;
+  const int levels_per_page = 100000;
+  // Vectors filled with random rep/defs and values to make pages.
+  std::vector<int32_t> values;
+  std::vector<int16_t> def_levels;
+  std::vector<int16_t> rep_levels;
+  std::vector<uint8_t> data_buffer;
+  std::vector<std::shared_ptr<Page>> pages;
+  MakePages<Int32Type>(&descr, num_pages, levels_per_page, def_levels, rep_levels, values,
+                       data_buffer, pages, Encoding::PLAIN);
+
+  // Vectors to read the values into.
+  std::vector<int32_t> read_values(batch_size, -1);
+  std::vector<int16_t> read_defs(batch_size, -1);
+  std::vector<int16_t> read_reps(batch_size, -1);
+
+  while (state.KeepRunning()) {
+    state.PauseTiming();
+    std::unique_ptr<PageReader> pager;
+    pager.reset(new test::MockPageReader(pages));
+    std::shared_ptr<ColumnReader> column_reader =
+        ColumnReader::Make(&descr, std::move(pager));
+    Int32Reader* reader = static_cast<Int32Reader*>(column_reader.get());
+    int values_count = -1;
+    state.ResumeTiming();
+    while (values_count != 0) {
+      if (skip == 1) {
+        DoNotOptimize(values_count = reader->Skip(batch_size));
+      } else {
+        int64_t values_read = 0;
+        DoNotOptimize(values_count = reader->ReadBatch(batch_size, read_defs.data(),
+                                                       read_reps.data(),
+                                                       read_values.data(), &values_read));
+      }
+    }
+  }
+}
```

Comment:
One nice thing to add is custom counters to make benchmark arguments more intelligible.
https://github.com/google/benchmark/blob/main/docs/user_guide.md#custom-counters

Something like:
```c++
  state.counters["batch_size"] = batch_size;
  state.counters["repetition"] = static_cast<int>(repetition);
```

Reply:
Or you can also use the undocumented `ArgNames` method:
https://github.com/google/benchmark/blob/398a8ac2e8e0b852fa1568dc1c8ebdfc743a380a/include/benchmark/benchmark.h#L1034-L1036

Reply:
Done

Reply:
Done